org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/xsd/participants/XSDRenameParticipant.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.lsp4xml.commons.BadLocationException;
+import org.eclipse.lsp4xml.dom.DOMAttr;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.dom.DOMElement;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.extensions.xsd.utils.XSDUtils;
+import org.eclipse.lsp4xml.services.extensions.IRenameParticipant;
+import org.eclipse.lsp4xml.services.extensions.IRenameRequest;
+import org.eclipse.lsp4xml.utils.DOMUtils;
+import org.eclipse.lsp4xml.utils.XMLPositionUtility;
+
+/**
+ * XSD rename
+ * 
+ */
+public class XSDRenameParticipant implements IRenameParticipant {
+
+	@Override
+	public void doRename(IRenameRequest request, List<TextEdit> locations) {
+		DOMDocument xmlDocument = request.getDocument();
+
+		if (!DOMUtils.isXSD(xmlDocument)) {
+			return;
+		}
+		for (TextEdit textEdit: getRenameTextEdits(request)) {
+			locations.add(textEdit);
+		}
+
+	}
+
+	private List<TextEdit> getRenameTextEdits(IRenameRequest request) {
+		DOMDocument document = request.getDocument();
+		DOMElement element = request.getElement();
+		Position position = request.getPosition();
+		String newText = request.getNewText();
+
+		if (XSDUtils.isXSComplexType(element)) {
+			List<DOMAttr> attributes = element.getAttributeNodes();
+			DOMAttr attrForValueRename = getAttributeForValueRename(document, attributes, position);
+
+			if (attrForValueRename == null) {
+				return Collections.emptyList();
+			}
+			
+			if (attrForValueRename.getName().equals("name")) {
+				List<Location> locations = getReferenceLocations(element);
+				return renameAttributeValueTextEdits(document, attrForValueRename, newText, locations);
+			}
+		}
+
+		return Collections.emptyList();
+	}
+
+	/**
+	 * Returns the DOMAttr in attributes whose range overlaps with renamePosition
+	 * @param document
+	 * @param attributes List of attributes to consider
+	 * @param renamePosition Position of interest
+	 * @return
+	 */
+	private DOMAttr getAttributeForValueRename(DOMDocument document, List<DOMAttr> attributes, Position renamePosition) {
+		try {
+			int renameLocationOffset = document.offsetAt(renamePosition);
+			for (DOMAttr attr : attributes) {
+				DOMNode attrValue = attr.getNodeAttrValue();
+
+				if (DOMNode.isIncluded(attrValue, renameLocationOffset)) {
+					return attr;
+				}
+			}
+		} catch (BadLocationException e) {
+
+		}
+
+		return null;
+	}
+
+	private List<Location> getReferenceLocations(DOMNode node) {
+
+		List<Location> locations = new ArrayList<>();
+
+		XSDUtils.searchXSOriginAttributes(node,
+		(origin, target) -> locations.add(XMLPositionUtility.createLocation(origin.getNodeAttrValue())),
+		null);
+
+		return locations;
+	}
+
+	private List<TextEdit> renameAttributeValueTextEdits(DOMDocument document, DOMAttr attribute, String newText, List<Location> locations) {
+		DOMNode attrValue = attribute.getNodeAttrValue();
+		List<TextEdit> textEdits = new ArrayList<>();
+		String textToInsert = "\"" + newText + "\"";
```

Comment:
the text to insert could have a xmlns:

For instance if you rename `paymentMethodType` to `XXXX` https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/test/resources/xsd/invoice-ns.xsd#L45 it should rename to `i:XXXX` at 
https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/test/resources/xsd/invoice-ns.xsd#L42